public/modules/chainManager.js

```diff
@@ -217,31 +219,23 @@ class ChainManager {
           // Wait for the app to start
           await new Promise((resolve, reject) => {
             childProcess.on('exit', async (code) => {
+              console.log(`[${chainId}] BitWindow exited with code ${code}`);
+
               if (code === 0) {
-                // Check if BitWindow is actually running using AppleScript
-                const checkProcess = spawn('osascript', ['-e', 'tell application "System Events" to count processes whose name is "bitwindow"']);
-                const isRunning = await new Promise((resolve) => {
-                  checkProcess.stdout.on('data', (data) => {
-                    resolve(parseInt(data.toString().trim()) > 0);
-                  });
-                  checkProcess.on('error', () => resolve(false));
-                });
+                const isRunning = await this.bitWindowClient.waitForConnection();
```

Comment:
If you start doing this, you should also use bitwindows brand new health service! The current `waitForConnection()` also requires bitcoin core to run, for example

Reply:
Thanks, fixed!